lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,172 @@
+
+- name: Enable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+  register: dkim_results
+
+- name: Example how to aws_ses_identity_dkim results to set DNS records for validation (if you use Route53 service for DNS)
+  route53:
+    record: "{{ item }}._domainkey.example.com"
+    value: "{{ item }}.dkim.amazonses.com"
+    type: CNAME
+    zone: example.com
+  with_items: "{{ dkim_results.dkim_attributes.dkim_tokens }}"
+
+- name: Disable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+    state: disabled
+'''
+
+RETURN = '''
+dkim_attributes:
+    description: Dictionary with DKIM information
+    type: complex
+    returned: always
+    sample: {
+        "dkim_tokens": [],
+        "dkim_enabled": "True",
+        "dkim_verification_status": "Pending"
+    }
+    contains:
+        dkim_tokens:
+            descritpion: Array of the 3 DKIM verification tokens needed for DNS records
+            type: list
+            sample: ['u5d5fa5twxzdni4aaoffdqvgnvku5kvu', '6xzw7n3abimypkyan5s6kuse4nymrfo2', 'nkwigx63eervm4hyja2lwzhjcfqyhka7']
+        dkim_enabled:
+            description: Boolean indicating if DKIM email sending is enabled or disabled for identity
+            type: bool
+            sample: True
+        dkim_verification_status:
+            description: Status of the DKIM verification process (DNS verification)
+            type: str
+            sample: 'Pending'
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def get_identity_dkim_settings(module, client, identity):
+    try:
+        response = client.get_identity_dkim_attributes(Identities=[identity])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to retrieve identity DKIM settings for {identity}'.format(identity=identity))
+    dkim_attributes = response['DkimAttributes']
+
+    return dkim_attributes[identity]
+
+
+def ses_verify_dkim_domain(module, client, identity):
+    domain = identity.split('@')[-1]
+    try:
+        if not module.check_mode:
+            response = client.verify_domain_dkim(Domain=domain)
+            return response['DkimTokens']
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to start DKIM verification for {domain}.'.format(domain=domain))
+
+
+def enable_identity_dkim_settings(module, client, identity):
+    try:
+        if not module.check_mode:
+            return client.set_identity_dkim_enabled(DkimEnabled=True, Identity=identity)
+    except (BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Failed to enable DKIM for {identity}.'.format(identity=identity))
+    except (ClientError) as e:
+        pass
```

Comment:
It would be really nice if we could limit this to a specific error code or something. I can't see anything in the AWS docs about a specific error for this case. I'll have a poke and see if I can identiify a way to do this.

Reply:
@stefanhorning I can't replicate the behaviour descibed in this comment. Runnig the tests without this catch seems to work fine. I wonder if AWS has updated the API to not error in this case. Can you chec if this catch and pass is still necessary?

Reply:
Couldn't reproduce it either anymore. So I removed this.